pkgs/stdenv/linux/default.nix

```diff
@@ -1,8 +1,55 @@
 # This file constructs the standard build environment for the
-# Linux/i686 platform.  It's completely pure; that is, it relies on no
+# Linux platform.  It's completely pure; that is, it relies on no
 # external (non-Nix) tools, such as /usr/bin/gcc, and it contains a C
 # compiler and linker that do not search in default locations,
 # ensuring purity of components produced by it.
+#
+# It starts from prebuilt seed bootstrapFiles and creates series of
+# nixpkgs instances (stages) to gradually rebuilt bootstrapFiles
+# with nixpkgs equivalents of packages.
+#
+# Goals of the bootstrap process:
```

Comment:
```suggestion
# Goals of the bootstrap process:
# 0. final stdenv must not reference any of the bootstrap files.
```
This is a separate goal from goal number 1 (on the next line), and stdenv already has achieved the first one.  Importantly, `allowedRequisites` provides an automated check that guarantees we won't accidentally lose this goal without noticing.

I fully support your more-ambitious goal (from #187225) of getting rid of the silly "copy libgcc_s hack".  If you manage to accomplish that, do you have any ideas on how we could add an automated check that would prevent people from copying things out of the bootstrap-files into stdenv?